include/dmtcp.h

```diff
@@ -46,6 +46,13 @@
 /* Define to the version of this package. */
 #define DMTCP_PLUGIN_API_VERSION "3"
 
+#ifdef __cplusplus
+namespace dmtcp {
+// Used by 'DMTCP_RESTART_PAUSE(level)', below; Defined in threadlist.cpp.
+extern int restartPauseLevel;
+}
+#endif // ifdef __cplusplus
+
```

Comment:
@karya0 , There's no special reason for the DMTCP_RESTART_MACRO definition to be inside `include//dmtcp.h`, and exposed to end users.  I believe that you had added this in `include/dmtcp.h`, in commit 485b5232b .

So, rather than rename it and expose it to end users, I would argue for continuing to hide this macro from end users.  They don't need it.

Currently, `DMTCP_RESTART_PAUSE` is found in threadlist.cpp and pluginmanager.cpp.  (And the occurrence in mtcp_restart.c defines a different expansion of that macro.)

So, let's hide the macro.  Please note that from earliest times, threadlist.cpp defines `dmtcp::restartPauseLevel`.  What's new is that we wish to reffer to this same global variable in `pluginmanager.cpp`.  Can you suggest a good include file to include the declaration of this global variable for DMTCP, and hide it from end users?  Thanks.